### PR TITLE
[GSoC] Generate nsw for signed integers operations

### PR DESCRIPTION
**Pull request by coodie**
Couple of months ago I have added 'nsw' (meaning undefined behaviour when overflowing on signed operations instead of wrapping) keyword to add instruction in LLVM IR since it was related to vectorization not occuring.  According to #6160 discussion there was a little inconsistency until this PR, because multiplications and subtractions would wrap while adds wouldn't, this tiny PR fixes that.

Proposed reviewers: @mppf and @dmk42

### CODE REVIEW

**Comment by dmk42** [comment]
This looks good.  I think the same thing should also be done with the CreateShl and CreateNeg calls.

File: test/llvm/generate-nsw-for-signed/SKIPIF [hunk 1]
```diff
@@ -0,0 +1 @@
+CHPL_
```
Comment (mppf) [inline test/llvm/generate-nsw-for-signed/SKIPIF]: Pretty sure this SKIPIF isn't finished... probably meant to be CHPL_LLVM!=llvm

**Comment by mppf** [comment]
Passed full local testing, with the SKIPIF fix

**Comment by dmk42** [comment]
I retested this after the additional nsws were added.  I got the following results.

* Passes full local testing without `--fast` on linux64
* Passes release/examples with and without `--fast` on a Mac
* Passes almost all of full local testing with `--fast` on linux64

The only test that fails with `--fast` is spec/marybeth/casts.  I believe that test is improper because it tries to use undefined behavior to detect undefined behavior, which cannot be relied upon.  I have asked the Chapel team if we can remove the undefined behavior part of that test.  Once that problem is resolved, this PR will be ready to merge.

**Pull request merged**